aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -323,6 +323,14 @@ func resourceAwsLambdaEventSourceMappingRead(d *schema.ResourceData, meta interf
 		return fmt.Errorf("error setting topics: %w", err)
 	}
 
+	if eventSourceMappingConfiguration.StartingPosition != nil {
+		d.Set("starting_position", eventSourceMappingConfiguration.StartingPosition)
+	}
```

Comment:
This makes change backwards compatible; so TF won't try to recreate event source mappings that currently have `nil` value on state. Maybe it's better to remove the `if` and allow value to get empty string? That would then cause a recreation of the resource.

Reply:
We should be safe as currently differences between `null` and the default value `""` are suppressed.
I will change this during merge.

Reply:
Perfect, thanks much!